akka-actor/src/main/scala/akka/io/SelectionHandler.scala

```diff
@@ -257,7 +257,12 @@ private[io] class SelectionHandler(settings: SelectionHandlerSettings) extends A
                               decision: SupervisorStrategy.Directive): Unit =
         try {
           val logMessage = cause match {
-            case e: ActorInitializationException if e.getCause ne null ⇒ e.getCause.getMessage
+            case e: ActorInitializationException if (e.getCause ne null) && (e.getCause.getMessage ne null) ⇒ e.getCause.getMessage
+            case e: ActorInitializationException if (e.getCause ne null) ⇒
+              e.getCause match {
+                case ie: java.lang.reflect.InvocationTargetException ⇒ ie.getTargetException.toString
+                case t: Throwable                                    ⇒ t.getClass.getSimpleName
```

Comment:
I’m not sure whether we have problematic exception types, but in general `getSimpleName` is not reliable when faced with Scala classes. Please use either `.getName` or `Logging.simpleName(t)`.